### DISCUSSION TITLE
Suggestion on the way of getting additional information / logs from cog/webkit

### DISCUSSION THREAD

**Discussion by pavlumik** [Igalia/cog#390] [Q&A] [answered]
Hello,

I am trying to integrate cog-0.12.0 and wpewebkit-2.34.2 into custom Linux image prepared with the help of Yocto and meta-webkit layer (https://layers.openembedded.org/layerindex/branch/hardknott/layer/meta-webkit/). Weston is used as an implementation of a Wayland compositor. 
In such environment I have got cog/wpewebkit  working on Raspberry Pi 4 HW and now trying to test similar configuration on Nvidia Jetson Xavier NX  HW (application level configuration is the same).

During my tests on Nvidia Jetson HW I noticed:
 - when starting cog with DRM (cog -P drm) the web pages rendered on the screen in spite of some error reports related to MESA / Gstreamer initialization;
 - when starting cog with Wayland platform (cog -P wl), there are no error reports in the terminal with cog application, however web pages are not rendered. 

Can you suggest (maybe there is some description on WIKIs?) how to activate logs in cog / WPE backend and understand if this is a cog issue or something else? 

One more question related to cog, I noticed that there are certain command line parameters enabling webkit features, for example if I start cog like:
/usr/bin/cog   --enable-javascript=1  --enable-javascript-markup=1  --javascript-can-open-windows-automatically=1 file://path/to/some/webpage.html
I would expect that Javascript code on the HTML5 webpage will be interpreted. Somehow simple JS alert popup message is not shown on the web page as a reaction of onclick handler for button. Did I miss some other enablers in command line?

Thanks in advance!

**Comment by philn** [comment, marked as answer]
I don't have much experience with NVIDIA, maybe someone else can help you about this.

> Somehow simple JS alert popup message is not shown on the web page as a reaction of onclick handler for button

Cog has some limited support to render option menus in Wayland platforms, but I think JS alert popups are not supported in WPE currently.

**Reply by pavlumik** [reply]
> Cog has some limited support to render option menus in Wayland platforms, but I think JS alert popups are not supported in WPE currently.

Thanks for quick reply! Do you know if there is some document where described list of supported features in WPE? I can see https://webkit.org/status/ and to my mind JS features here fully supported by WebKit itself...

**Reply by philn** [reply]
I'm not aware of a WPE-specific doc. WPE doesn't depend on any widget library, so supporting popups and similar UI components is not trivial. As I mentioned earlier, there is some support for popups in WPE/Cog, but JS alerts is likely unsupported at this time. Patch welcome!
